CMakeLists.txt

```diff
@@ -22,7 +22,7 @@ option(BUILD_TESTING_INTEGRATION "" OFF)
 include(CTest)
 
 set(CMAKE_C_STANDARD 11)
-set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
Because you changed this to `17`, you have to set the `cmake_minimum_required(VERSION 3.8)`, because version `3.8` of `CMake` is the first that supports C++17.
https://cmake.org/cmake/help/v3.8/prop_tgt/CXX_STANDARD.html
https://cmake.org/cmake/help/v3.7/prop_tgt/CXX_STANDARD.html